ST_mortality.c

```diff
@@ -268,106 +278,62 @@ void (*simulatePrescribedFire)(GrpIndex, RealF) = _simulatePrescribedFire;
  * \ingroup MORTALITY
  */
 void mort_EndOfYear(void) {
-    GrpIndex rg;
-    GroupType *g = NULL;
-    SppIndex sp;
-    IntU j;
-    RealF fire_possibility, random_number, biomass_cheatgrass;
-    char *cheatgrass_name = "brte";
-    int i = 0;
-    Bool prescribed_fire_on = FALSE;
-
-    /* Check species index number from the beginning to all the species in
-     *  species.in , if the species name == checkname then get the biomass and stop the loop*/
-    for (i = 0; i < Globals->sppCount; i++) { /* if species name = checkname = brte then get the biomass of brte(cheatgrass)*/
-        if (strcmp(cheatgrass_name, Species[i]->name) == 0) {
-            biomass_cheatgrass = Species_GetBiomass(i); /* calculate biomass of cheatgrass*/
-            g = RGroup[Species[i]->res_grp];
-            break;
-        }
-    }
+  GrpIndex rg;
+  SppIndex sp;
+  IntU j;
+  RealF cheatgrassBiomass;
+
+  /* Update the precipitation parameters that determine cheatgrass-driven
+   * wildfire. */
+  _updateCheatgrassPrecip(Globals->currYear);
+  /* printf("%d:\tMeanSpring = %f\t ThisSpring = %f\t MeanWinter = %f\t LastWinter = %f\n",
+         Globals->currYear, cheatgrassPrecip->springMean, cheatgrassPrecip->currentSpring, 
+         cheatgrassPrecip->winterMean, cheatgrassPrecip->lastWinter); */
+
+  /* Reset prescribedfire and wildfire for every RGroup. */
+  ForEachGroup(rg){
+    RGroup[rg]->prescribedfire = 0;
+    RGroup[rg]->wildfire = 0;
+  }
 
-    /* Set a random number outside of the loop to make sure the kill probability for each functional group is the same */
-    random_number = RandUni(&mortality_rng);
+  if(UseCheatgrassWildfire) {
+    cheatgrassBiomass = _getCheatgrassBiomass();
+    // If cheatgrass was found in the Species array.
+    if(cheatgrassBiomass >= 0) {
+      _simulateWildfire(cheatgrassBiomass);
+    }
+  } else {
```

Comment:
if cheatgrassBiomass = 0, what happens? It looks like cover = 0 and then wildfire never occurs. There there is no point of going through this function at all. Unless I am missing something?

Reply:
@kpalmqui I decided to let the function run even if there is 0 cheatgrass biomass in case someone down the line decides to change the algorithm, but I can modify the equality if you would like!

Reply:
@chaukap do you think this is actually faster than having the code skip this, but then proceed to prescribed fire? I guess we can leave it as it is for now.